src/git.ts

```diff
@@ -295,3 +301,34 @@ export function removeGitReposFromStore(folderNames: string[]) {
         folderDelete(dir, {debugLog: false});
     });
 }
+
+const lsRemote = async (url: string) => {
+  return new Promise((resolve) => {
+    // timeout for sanity - detached to block stdin - stdio: ignore to prevent stdout from blocking
+    const child = childProcess.spawn('git', ['ls-remote', url], { timeout: 10_000, detached: true, stdio: 'ignore' })
+    // if the process will need to ask for username+password from stdin it will get an error and exit with error code != 0
+    child.on('close', code => {
+      resolve(code === 0)
+    })
+  })
+}
+
+export async function canAuthGitRepo(repoUrl: string): Promise<{ querySuccessful: boolean, protocol: GitProtocols }> {
+  const sshFormat = convertUrlToProtocol(repoUrl, GitProtocols.SSH);
+  const httpFormat = convertUrlToProtocol(repoUrl, GitProtocols.HTTPS);
+
+  if (await lsRemote(sshFormat)) {
```

Comment:
Looking at our DB I think we should try https first

Reply:
Our DB has the repo URLs that were used in the CI/CD's so we can't exactly know which protocol the clients use on their local machine.  
Second, if SSH is configured, everything will "just work".  
With HTTP you have multiple choices and managers that hook with the git client that can prompt or provide access.  
For example, if you run git commands (on HTTP remote) from within your VSCode terminal - VSCode injects `GIT_ASKPASS` env var which redirects the git CLI to ask for username/password from VSCode itself, who might prompt the user for credentials.  
Another approach is the git credential.helper